### ISSUE TITLE
German CoNLL-2003 data

### ISSUE THREAD

**Issue by stefan-it** [deepset-ai/FARM#235]
Hi,

it seems that the `DOWNSTREAM_TASK_MAP` uses the wrong data for the German CoNLL-2003.

I had a look at the data that is specified e.g. for the `conll03detrain` task and I think they also include the data from the GermEval shared task.

I compared the original CoNLL-2003 dataset (created from Reuters CD) and the correct data is located in this [conll2003 folder](https://github.com/MaviccPRP/ger_ner_evals/tree/master/corpora/conll2003). Files are `deu.train`, `deu.testa` and `deu.testb`.

However, they need a few preprocessing steps: conversion from iso-latin to utf-8 and the original dataset uses the IOB1 tagging scheme.

I think we should use the original dataset for further benchmarks :)

**Comment by Timoeller** [comment]
Thanks for the heads up.
We will double check and let you know asap.

Do you maybe know another site that hosts the correctly transformed data? We would prefer to not host the copyrighted conll03 data ourselves.
As last resort we will do the necessary transformations on FARM side.

**Comment by brandenchan** [comment]
Hey thanks for pointing this out. You are actually right - the conll dataset has at least some if not all of germeval14 in it. We are looking more deeply in to it now and will get back to you.

**Comment by stefan-it** [comment]
@brandenchan at the moment I couldn't find the CoNLL data for German that is both in utf-8 and iob2. So I think we need to implement some function for encoding and fixing the tagging format.

Here's a snippet for IOB1 to IOB2 (haven't test it yet), that we could use: [Gist](https://gist.github.com/allanj/b9bd448dc9b70d71eb7c2b6dd33fe4ef) :)

**Comment by brandenchan** [comment]
I was just about to ask you for that code - thanks! Will keep you updated on our progress

**Comment by brandenchan** [comment]
Hey, I actually found a bug in the code. From what I can tell now, the links to CoNLL and Germeval datasets point to the right datasets. The first sentence in the CoNLL dataset starts "Schartau sagte dem..." and the first sentence in the Germeval dataset is "Ereignis und Erzählung oder..."

I could not find any overlapping sentences between the datasets. Can you provide any examples of this?

Also it seems that the dev set of Germeval that is linked in FARM is simply a slice of the train set and I propose that we remove the link to this file.

**Comment by Timoeller** [comment]
Hey @stefan-it 
I had time to look into this issue and can confirm this huge bug!

The data that was downloaded as ["conll03de"](https://raw.githubusercontent.com/MaviccPRP/ger_ner_evals/master/corpora/training_data_for_Stanford_NER/NER-de-train-conll-formated.txt) in FARM is text from more recent newspapers articles. I found examples pointing to n-tv.de articles from 2008 and a rp-online article from 2009. So our "conll03de" version cannot be the real deal.

I created a PR #248 with your proposed changes.

We updated our evaluation of the German Bert vs multilingual models in our [blog post](https://deepset.ai/german-bert) as well. The post about XLM-R will be updated soon, too.

Thanks for the hint! This was a big mess on our side : )
